### DISCUSSION TITLE
Support for Elecrow ESP Terminal? (ESP32-S3)

### DISCUSSION THREAD

**Discussion by Underestimated** [HASwitchPlate/openHASP#492] [Hardware]
Hi team,
Hoping this is the right way to request investigation.

I've gotten my hands on an Elecrow ESP Terminal with 3.5inch SPI Capacitive Touch Display. 
( https://littlebirdelectronics.com.au/products/esp-terminal-3-5-inch-480x320-spi-tft-capacitive-touch-display-with-ov2640-camera ) 
and I want to try getting openHASP to run on it.

It looks like the current releases don't have support for it yet, but I've seen you're implementing the Makerfabs MaTouch_ESP32-S3 in a near release, which is related.

This board runs off a ESP32-S3 module, with a ILI9488 Capacitive Touch Screen on a FT6236 touch controller. (16mb flash 8mb psram) it has a camera module and header but I've removed that as I'm really interested only in the touchscreen.

Seeing as you're working on a board that isn't far from the specs of this one, I wanted to see if you could implement this one as well, so I can give it a shot?

The wiki with schematics and pinouts for the board is at https://www.elecrow.com/wiki/index.php?title=ESP_Terminal_with_3.5inch_SPI_Capacitive_Touch_Display

Either way thanks in advance!

**Comment by fvanroie** [comment]
Thanks for bringing the ESP Terminal to our attention, these devices will be added soon.

**Reply by gromgsxr** [reply]
Did this ever get added? I have one of the rgb non camera versions and would be a nice addition

**Reply by fvanroie** [reply]
You can find it from Github under the Actions tab. Click on the last successful build and scroll down to download the `elecrow.zip` file from the asset list.

**Reply by bwims** [reply]
Hi,

I'm a bit of a Github noob.  Could you explain in a bit more detail where the elecrow.zip file is for someone like me?  I went to the actions tab and got totally lost.

Thanks!

B.

**Reply by fvanroie** [reply]
At the bottom of each successful run you find a list of Artifacts i.e. zip files containing the firmwares per manufacturer. You download the elecrow.zip at the bottom of the [last run](https://github.com/HASwitchPlate/openHASP/actions/runs/7561810549).

**Reply by bwims** [reply]
Hello,

Progress! I used esp32-terminal-rgb_full_16MB_v0.7.0-rc9_c200edf.bin and installed it using the esphome.io web flasher.
I got the wifi configuration screen and configured my wifi details.

The next screen I got displayed "plate" in the top left corner but nothing else.

Did I use the correct bin file?

[This ](https://www.elecrow.com/esp-terminal-with-esp32-3-5-inch-parallel-480x320-tft-capacitive-touch-display-rgb-by-chip-ili9488.html)is the device I am using so I think it's the correct bin file.

Any idea of what went wrong?

Thanks!

B.

**Reply by fvanroie** [reply]
Nothing's wrong. `Plate` is just a placeholder, now you need to upload a `pages.jsonl` file to setup your device.

**Comment by rasel99** [comment]
I have the same device Elecrow ESP Terminal with 3.5inch SPI Capacitive Touch Display
https://www.elecrow.com/esp-terminal-3-5-inch-320-480-spi-tft-capacitive-touch-display-with-ov2640-camera.html

and was able flash it successfully using [esp32-terminal-spi_full_16MB_v0.7.0-rc9_33d21e0.bin](https://nightly.openhasp.com/firmware/esp32-terminal-spi_full_16MB_v0.7.0-rc9_33d21e0.bin)

Everything works as intended.

Would it be possible to add camera support and display the camera's stream directly on the screen?

**Reply by Lateralleap** [reply]
Hi @rasel99 

Would you be able to share the pin assignments you used in your config for:
DC
RESET
SPI CLK
SPI MOSI - 13?
SPI MISO - 14?

The documentation that came with the terminal was a little unclear

Thanks

**Reply by fvanroie** [reply]
@rasel99 Can you please share your config file?

**Comment by Lateralleap** [comment]
Thanks for following up. I thought the case was closed

I have since returned the device
